npm/private/lifecycle/lifecycle-hooks.js

```diff
@@ -34,26 +34,16 @@ async function makeBins(nodeModulesPath, scope, segmentsUp) {
             'package.json'
         )
         if (fs.existsSync(packageJsonPath)) {
-            let packageJson = ''
-
-            // Catch invalid JSON errors and retry or throw a more detailed Error.
-            // See: https://github.com/aspect-build/rules_js/issues/483
-            const json = await fs.promises.readFile(packageJsonPath)
+            let packageJson
             try {
-                packageJson = JSON.parse(json)
+                packageJson = JSON.parse(
+                    await fs.promises.readFile(packageJsonPath)
+                )
             } catch (e) {
-                console.error(
-                    `Error parsing ${packageName}/package.json: ${e}\n\n""""\n${json}\n""""\nRetrying...`
+                // Catch and throw a more detailed error message.
+                throw new Error(
+                    `Error reading ${packageName}/package.json: ${e}\n\n""""\n${json}\n""""`
```

Comment:
json will be undefined now

Reply:
You can keep the readFile out of the try/catch, it will throw and the error is sound.
try/catch around the json.parse is sufficient. Perhaps change the error message to `Error parsing`?